x/crosschain/genesis.go

```diff
@@ -36,24 +34,22 @@ func InitGenesis(ctx sdk.Context, k keeper.Keeper, genState types.GenesisState)
 		}
 	}
 
-	// Set all the chain nonces
-
 	// Set all the last block heights
 	for _, elem := range genState.LastBlockHeightList {
 		if elem != nil {
 			k.SetLastBlockHeight(ctx, *elem)
 		}
 	}
 
-	// Set all the cross-chain txs
-	tss, found := k.GetObserverKeeper().GetTSS(ctx)
-	if found {
-		for _, elem := range genState.CrossChainTxs {
-			if elem != nil {
-				k.SaveCCTXUpdate(ctx, *elem, tss.TssPubkey)
-			}
+	// Set the cross-chain transactions only,
+	// We don't need to call SaveCCTXUpdate as the other fields are being set already
+	for _, elem := range genState.CrossChainTxs {
+		if elem != nil {
+			cctx := *elem
+			k.SetCrossChainTx(ctx, cctx)
```

Comment:
While working on this PR, I realized that We were saving many different parts of the state and doing calculations during genesis. 
I have the opinion now that the init genesis should be as simple as possible, and we should only be importing the exported state without modifying it. 
I am open to hearing other thoughts and opinions.

Save update did the following 
SetNonceToCCTX - Part of the observer state and should be managed by observer module 
SetCrossChainTx - We are still doing this 
UpdateInboundHashToCCTX - This is already being set separately using `genState.InboundHashToCctxList`
UpdateZetaAccounting - This can be set using genstate.ZetaAccounting , (This was remove in this pr https://github.com/zeta-chain/node/pull/3230/files#diff-2dce2b3ecc78bbb08e9523c64c1ad6ec82694ea0d821d032570383d951c5989f , but I think it makes more sense to keep it )

Reply:
Yes, it's correct, genesis logic should only use getter/setter function but it seems it would makemore sense to create a dedicated PR for these changes

Reply:
Happy to create a separate pr for this , and merge that before trying to merge this if needed .
However since it does nto change any logic, I considered putting it in this pr directly , but if it helps with reviewing , I can create a pr